### PR TITLE
RLPNC-7516: Add Score_if_null to Java Bindings for record-similarity

### PR DESCRIPTION
**Pull request by fhasanaj**

### CODE REVIEW

File: model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/RecordSimilarityFieldInfo.java [hunk 1]
```diff
@@ -29,6 +29,13 @@
...
 public class RecordSimilarityFieldInfo {
+    // The default setting for scoreIfNull, which disables this feature
+    private static final double DEFAULT_SCORE_IF_NULL = -1.0;
     @NotNull @Valid RecordFieldType type;
     Double weight;
+    /**
+     * the score that this field should return if the field is null in a record
+     */
+    @Builder.Default
+    @NotNull @Valid Double scoreIfNull = DEFAULT_SCORE_IF_NULL;
```
Comment (iredpath) [inline model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/RecordSimilarityFieldInfo.java]: I know this matches what we do in other parts of the code, but can we just have the default "off" value be `null` instead?  That way we don't need any additional annotations to ignore default values during serialization

**Comment by ethteck** [comment]
Review is all good on the names side

**Comment by ethteck** [comment]
@seth-mg @asoosprecognox aside from the merge conflict, does this look alright to you? If I should reach out to someone else to get this reviewed, let me know

**Comment by asoosprecognox** [comment]
Maybe add a scoreIfNull in the test that is a value and not just null, if that makes sense with the business logic. Otherwise it looks good if sonar is happy (no major/critical issues) after the merge conflicts are resolved

File: model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/RecordSimilarityFieldInfo.java [hunk 2]
```diff
@@ -31,4 +31,8 @@
 public class RecordSimilarityFieldInfo {
```
Comment (iredpath) [inline model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/RecordSimilarityFieldInfo.java:33]: We actually probably wanted this already, but this should have `@JsonInclude(JsonInclude.Include.NON_NULL)` so that null weights/scoreIfNull values don't show up in the response

**Comment by iredpath** [comment]
@fhasanaj there's another merge conflict now after the RLPNC-7512 merge

**Comment by iredpath** [comment]
LGTM, @asoosprecognox or @seth-mg it would be great if one of you could take one more look before we merge this in

**Comment by fhasanaj** [comment]
@asoosprecognox or @seth-mg can we get an approval from one of you please

**Review by asoosprecognox — APPROVED** [review]
LGTM
